### PR TITLE
Include Application Signalling (HbbTV) PIDs

### PR DESCRIPTION
**Pull request by freddy36**
HbbTV data is sent in a extra stream (dedicated PID). To make it work we have to include the PID signalled via the PMT stream_type 0x05 (ITU-T Rec. H.222.0 | ISO/IEC 13818-1 private_sections) entry with application_signalling_descriptor set.

### CODE REVIEW

File: src/autoconf_pmt.c [hunk 1]
```diff
@@ -278,10 +296,6 @@ void autoconf_get_pmt_pids(auto_p_t *auto_p, mumudvb_ts_packet_t *pmt, int *pids
 			break;
 
 			//Now, the list of what we drop
-		case 0x05:
-			log_message( log_module,  MSG_DEBUG, "Dropped PID %d, type : 0x05, ITU-T Rec. H.222.0 | ISO/IEC 13818-1 private_sections \n",pid);
-			continue;
-			//Digital Storage Medium Command and Control (DSM-CC) cf H.222.0 | ISO/IEC 13818-1 annex B
```
Comment (braice) [inline src/autoconf_pmt.c:284]: Could you keep this comment ?

File: src/autoconf_pmt.c [hunk 2]
```diff
@@ -227,9 +227,27 @@ void autoconf_get_pmt_pids(auto_p_t *auto_p, mumudvb_ts_packet_t *pmt, int *pids
...
+		case 0x05: /* ITU-T Rec. H.222.0 | ISO/IEC 13818-1 private_sections */
+			if(descr_section_len) //If we have an associated descriptor, we'll search information in it
+			{
+				if(pmt_find_descriptor(0x6f,pmt->data_full+i+PMT_INFO_LEN,descr_section_len, NULL)){ // application_signalling_descriptor (AIT/HbbTV)
+					log_message( log_module,  MSG_DEBUG,"  Application Signalling \tPID %d\n",pid);
+					pid_type=PID_EXTRA_APPLICATION_SIGNALLING;
+					break;
+				}else
+				{
+					log_message( log_module,  MSG_DEBUG,"Unknown descriptor for stream_type 0x05 see EN 300 468 v1.9.1 table 12, PID %d descriptor tags : ", pid);
```
Comment (braice) [inline src/autoconf_pmt.c]: I would say something like dropped pid private section as below

**Review by braice — CHANGES_REQUESTED** [review]
Can you also please update the documentation

**Comment by freddy36** [comment]
Made the changes to the comment/log message.
Which doc do you thing needs to be updated? Changelog seems to be completely outdated and the README doesn't mention this PID filtering mechanism at all.

**Comment by braice** [comment]
Sorry for the late reply,

The doc to be update is the README_CONF

I was compiling the asciidoc before and it was getting out of date, now you
can see the blow directly at github (and the website points to it), so
despite the formatting issues at least it is up to date
https://github.com/braice/MuMuDVB/blob/mumudvb2/doc/README_CONF.asciidoc

**Comment by freddy36** [comment]
What part of the documentation do you want me to update?
The existing pid filtering details for autoconfiguration seems to be completely undocumented.
Why are certain PIDs droped at all in the current design? Did they cause problems with some end devices?

**Comment by braice** [comment]
Sorry I was confused, the patch is OK I am merging

**Pull request merged**